pkg/skaffold/build/cache/hash.go

```diff
@@ -38,15 +38,72 @@ import (
 
 // For testing
 var (
-	hashFunction           = cacheHasher
-	artifactConfigFunction = artifactConfig
+	newArtifactHasherFunc = newArtifactHasher
+	fileHasherFunc        = fileHasher
+	artifactConfigFunc    = artifactConfig
 )
 
-func getHashForArtifact(ctx context.Context, depLister DependencyLister, a *latest.Artifact, mode config.RunMode) (string, error) {
+type artifactHasher interface {
+	hash(ctx context.Context, a *latest.Artifact) (string, error)
+}
+
+type Artifacts map[string]*latest.Artifact
+
+type artifactHasherImpl struct {
```

Comment:
Could use this compile-time trick to ensure this type implements the function:
```
var _ artifactHasher = (*artifactHasherImpl)(nil)

Reply:
neat, but I don't think I need it here since I have the `newArtifactHasher` function which will also fail at compile-time.